pkg/issuer/acme/setup.go

```diff
@@ -565,3 +575,86 @@ var acmev1ToV2Mappings = map[string]string{
 	fmt.Sprintf("%s/", acmev1Prod):    acmev2Prod,
 	fmt.Sprintf("%s/", acmev1Staging): acmev2Staging,
 }
+
+func (a *Acme) validateDNSSolvers(ctx context.Context, issuer v1.GenericIssuer) []string {
+	var warning []string
+
+	secrets := extractSecrets(issuer)
+	if len(secrets) == 0 {
+		return warning
+	}
+
+	for _, s := range secrets {
+		res, err := a.secretsClient.Secrets(issuer.GetNamespace()).Get(ctx, s.Name, metav1.GetOptions{})
+		if err != nil {
+			warning = append(warning, fmt.Sprintf("failed to get secret %q: %v", s.Name, err))
+			continue
+		}
+
+		if s.Key != "" {
+			if _, ok := res.StringData[s.Key]; !ok {
```

Comment:
I would suggest using `.Data` instead of `.StringData` based on the fact that StringData could be empty on READ.

https://github.com/kubernetes/kubernetes/blob/c70011cf108c8af63c5b4af3c2f8ff88b333a698/staging/src/k8s.io/api/core/v1/generated.proto#L5771